tests/unit/src/kakarot/instructions/test_push_operations.py

```diff
@@ -11,101 +11,17 @@ async def push_operations(starknet: Starknet):
-        await push_operations.test__exec_push18_should_add_18_byte_to_stack().call()
-
-    async def test__exec_push19_should_add_19_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push19_should_add_19_byte_to_stack().call()
-
-    async def test__exec_push20_should_add_20_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push20_should_add_20_byte_to_stack().call()
-
-    async def test__exec_push21_should_add_21_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push21_should_add_21_byte_to_stack().call()
-
-    async def test__exec_push22_should_add_22_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push22_should_add_22_byte_to_stack().call()
-
-    async def test__exec_push23_should_add_23_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push23_should_add_23_byte_to_stack().call()
-
-    async def test__exec_push24_should_add_24_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push24_should_add_24_byte_to_stack().call()
-
-    async def test__exec_push25_should_add_25_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push25_should_add_25_byte_to_stack().call()
-
-    async def test__exec_push26_should_add_26_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push26_should_add_26_byte_to_stack().call()
-
-    async def test__exec_push27_should_add_27_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push27_should_add_27_byte_to_stack().call()
-
-    async def test__exec_push28_should_add_28_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push28_should_add_28_byte_to_stack().call()
-
-    async def test__exec_push29_should_add_29_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push29_should_add_29_byte_to_stack().call()
-
-    async def test__exec_push30_should_add_30_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push30_should_add_30_byte_to_stack().call()
-
-    async def test__exec_push31_should_add_31_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push31_should_add_31_byte_to_stack().call()
-
-    async def test__exec_push32_should_add_32_byte_to_stack(self, push_operations):
-        await push_operations.test__exec_push32_should_add_32_byte_to_stack().call()
+    @pytest.mark.parametrize('i', range(2, 33))
+    async def test__exec_push_should_raise(self, push_operations, i):
+        with kakarot_error():
+            await push_operations.test__exec_push_should_raise(i).call()
+
+    @pytest.mark.parametrize('i', range(1, 17))
+    async def test__exec_push_should_add_1_through_16_bytes_to_stack(self, push_operations, i):
+        await push_operations.test__exec_push_should_add_1_through_16_bytes_to_stack(i).call()
```

Comment:
so update this

```
peek = await push_operations.test_should_push(i).call()
assert peek = 256**i - 1
```

Reply:
How can I get the stack's last value here?

Reply:
you can return it in the cairo test func, can't you?

Reply:
Yea, I had to look into the struct of `StarknetCallInfo`. Wasn't aware of it.